src/hisp/helpers.py

```diff
@@ -0,0 +1,148 @@
+        """
+        current_time = 0
+        for i, row in enumerate(self.data):
+            nb_pulses = int(row[1])
+            phase_duration = nb_pulses * self.get_pulse_duration(i)
+            if t <= current_time + phase_duration:
+                return i
+            else:
+                current_time += phase_duration
+
+        raise ValueError(
+            f"Time t {t} is out of bounds of the scenario file. Maximum time is {self.get_maximum_time()}"
+        )
+
+    def get_pulse_type(self, t: float) -> str:
+        """Returns the pulse type as a string at time t.
+
+        Args:
+            t (float): time in seconds
+
+        Returns:
+            str: pulse type (eg. FP, ICWC, RISP, GDC, BAKE)
+        """
+        row_idx = self.get_row(t)
+        return self.data[row_idx][0]
+
+    def get_pulse_duration(self, row: int) -> float:
+        """Returns the total duration of a pulse in seconds for a given row in the file.
+
+        Args:
+            row (int): the row index in the scenario file
+
+        Returns:
+            float: the total duration of the pulse in seconds
+        """
+        row_data = self.data[row]
+        pulse_type = row_data[0]
+        if pulse_type == "RISP":  # hard coded because it's zero in the files
+            ramp_up = 10
+            steady_state = 250
+            ramp_down = 10
+            waiting = 1530
+            total_duration = ramp_up + steady_state + ramp_down + waiting
+            return total_duration
+
+        ramp_up = float(row_data[2])
+        steady_state = float(row_data[4])
+        ramp_down = float(row_data[3])
+        waiting = float(row_data[5])
+
+        total_duration = ramp_up + steady_state + ramp_down + waiting
+        return total_duration
+
+    def get_pulse_duration_no_waiting(self, row: int) -> float:
+        """Returns the total duration of a pulse in seconds for a given row in the file.
+
+        Args:
+            row (int): the row index in the scenario file
+
+        Returns:
+            float: the total duration of the pulse in seconds
+        """
+        row_data = self.data[row]
+        pulse_type = row_data[0]
+        if pulse_type == "RISP":  # hard coded because it's zero in the files
+            ramp_up = 10
+            steady_state = 250
+            ramp_down = 10
+            duration = ramp_up + steady_state + ramp_down
+            return duration
+
+        ramp_up = float(row_data[2])
+        steady_state = float(row_data[4])
+        ramp_down = float(row_data[3])
+
+        duration = ramp_up + steady_state + ramp_down
+        return duration
+    
+    def get_time_till_row(self, row:int) -> float:
+        """Returns the time that has elapsed in scenario up until input row.
+
+        Args:
+            row (int): the row index in the scenario file
+
+        Returns:
+            float: the time that has elapsed in scenario until input row 
+        """
```

Comment:
Just by looking at this it is unclear if this is inclusive or exclusive: does it include the "current" row? In other words does it count until the begining of the current `row` or the end of the current row